Source/WebCore/page/Navigation.h

```diff
@@ -136,7 +136,7 @@ class Navigation final : public RefCounted<Navigation>, public EventTarget, publ
     ExceptionOr<void> updateCurrentEntry(UpdateCurrentEntryOptions&&);
 
     bool dispatchTraversalNavigateEvent(HistoryItem&);
-    bool dispatchPushReplaceReloadNavigateEvent(const URL&, NavigationNavigationType, bool isSameDocument);
+    bool dispatchPushReplaceReloadNavigateEvent(const URL&, NavigationNavigationType, bool isSameDocument, SerializedScriptValue* classicHistoryAPIState = nullptr);
```

Comment:
Seems a little strange to pass through a pointer just to check that it’s null. Can this be a boolean instead?

Reply:
That is a good point, however we'll need the proper type to implement setting state:
https://html.spec.whatwg.org/multipage/nav-history-apis.html#fire-a-push/replace/reload-navigate-event (step 6).